state/executor.go

```diff
@@ -562,6 +567,21 @@ func (t *Transition) run(contract *runtime.Contract, host runtime.Host) *runtime
 		return t.deploymentAllowlist.Run(contract, host, &t.config)
 	}
 
+	if t.txnAllowList != nil {
```

Comment:
What happens if we have both `deployment` and `txn` allow lists defined, but an address is not in the `txn` allow list? Will this condition block him to deploy a contract? Correct me if I'm wrong, but if you are deploying a new contract, you will check the `txnAllowList` as well, and if you are not in that list, you'll be rejected.

Reply:
> What happens if we have both deployment and txn allow lists defined, but an address is not in the txn allow list? Will this condition block him to deploy a contract? 

Yes, it will block the user from deploying a contract (create transaction), all transactions follow this code path.

> Correct me if I'm wrong, but if you are deploying a new contract, you will check the txnAllowList as well, and if you are not in that list, you'll be rejected.

This is what's happening, I added a new test case to ensure it's like that.

Reply:
Yep, but is that something we want and expect? To me it's a little bit confusing. Maybe it's not a valid case to be in deployment allow list, but not in transaction allow list, but it can happen in theory right?

Reply:
Yes it can happen, but in security, the norm is that the most restrictive permission apply, so even if you are allowed to deploy but not allowed to send transactions, you shouldn't be able to send transactions.